pkg/encryption/decrypt_store.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package encryption
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"hash"
+
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"golang.org/x/crypto/sha3"
+)
+
+func hashFunc() hash.Hash {
+	return sha3.NewLegacyKeccak256()
+}
+
+type decryptingGetter struct {
+	storage.Getter
+}
+
+func NewDecryptingGetter(s storage.Getter) storage.Getter {
```

Comment:
i think thereshould be an encrypted store in storage rather. this pkg should not depend on storage